diff --git a/kernel/ksud.c b/kernel/ksud.c
index d2ae017a..b72ca385 100644
--- a/kernel/ksud.c
+++ b/kernel/ksud.c
@@ -527,6 +526,10 @@ static void stop_execve_hook()
 {
 	bool ret = schedule_work(&stop_execve_hook_work);
 	pr_info("unregister execve kprobe: %d!\n", ret);
+#ifdef CONFIG_KSU_SUSFS_SUS_SU
+	susfs_is_sus_su_ready = true;
+	pr_info("susfs: sus_su is ready\n");
+#endif
 }
 
 static void stop_input_hook()
