diff --git a/kernel/ksud.c b/kernel/ksud.c
index d2ae017a..b72ca385 100644
--- a/kernel/ksud.c
+++ b/kernel/ksud.c
@@ -52,7 +52,6 @@ static struct work_struct stop_execve_hook_work;
 static struct work_struct stop_input_hook_work;
 
 #ifdef CONFIG_KSU_SUSFS_SUS_SU
-bool ksu_devpts_hook = false;
 bool susfs_is_sus_su_ready = false;
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_SU
 
@@ -527,6 +526,10 @@ static void stop_execve_hook()
 {
 	bool ret = schedule_work(&stop_execve_hook_work);
 	pr_info("unregister execve kprobe: %d!\n", ret);
+#ifdef CONFIG_KSU_SUSFS_SUS_SU
+	susfs_is_sus_su_ready = true;
+	pr_info("susfs: sus_su is ready\n");
+#endif
 }
 
 static void stop_input_hook()
