diff --git a/kernel/sucompat.c b/kernel/sucompat.c
index 47a673cd..d9eba557 100644
--- a/kernel/sucompat.c
+++ b/kernel/sucompat.c
@@ -268,7 +268,6 @@
 	enable_kprobe(&execve_kp);
 	enable_kprobe(&newfstatat_kp);
 	enable_kprobe(&faccessat_kp);
-	enable_kprobe(&pts_unix98_lookup_kp);
 	ksu_devpts_hook = false;
 }
 
@@ -276,7 +275,6 @@
 	disable_kprobe(&execve_kp);
 	disable_kprobe(&newfstatat_kp);
 	disable_kprobe(&faccessat_kp);
-	disable_kprobe(&pts_unix98_lookup_kp);
 	ksu_devpts_hook = true;
 }
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_SU
